pkg/nsx/services/ippool/ippool.go

```diff
@@ -71,6 +72,11 @@ func InitializeIPPool(service common.Service) (*IPPoolService, error) {
 
 func (service *IPPoolService) CreateOrUpdateIPPool(obj *v1alpha2.IPPool) (bool, bool, error) {
 	nsxIPPool, nsxIPSubnets := service.BuildIPPool(obj)
+	for _, ipSubnet := range nsxIPSubnets {
+		if *ipSubnet.IpBlockPath == "" {
+			return false, false, util.IPBlockAllExhaustedError{Desc: "all ip blocks are exhausted"}
+		}
```

Comment:
if first  *ipSubnet.IpBlockPath == "" , does that means all ipblock are exhausted? 
should we iterate all ipsubnet?

Reply:
no need, if *ipSubnet.IpBlockPath == "", it represents all exhausted, it is the logic indeed.